plugins/widget/plugin.js

```diff
@@ -2180,8 +2225,39 @@
 			element = wrapper.getFirst( Widget.isDomWidgetElement );
 			// If wrapper contains widget element - unwrap it and wrap again.
 			if ( element.type == CKEDITOR.NODE_ELEMENT && element.data( 'widget' ) ) {
+				styleDefinition = { attributes: {}, element: element.getName() };
+
+				styles = CKEDITOR.tools.parseCssText( wrapper.getAttribute( 'style' ) );
+				if ( styles ) {
+					styleDefinition.styles = styles;
+				}
+
+				attributes = element.getAttributes();
+				for ( key in attributes ) {
+					if ( key !== 'style' && key !== 'class' && key.substring( 0, 4 ).toLowerCase() !== 'data' ) {
+						styleDefinition.attributes[ key ] = attributes[ key ];
+					} else if ( key === 'class' ) {
+						classes = attributes[ key ].split( ' ' );
+						classes = CKEDITOR.tools.array.filter( classes, function( item ) {
+							return item.substring( 0, 3 ) !== 'cke';
+						} );
+						classes = classes.join( ' ' );
+						if ( classes ) {
+							styleDefinition.attributes[ 'class' ] = classes;
```

Comment:
It is, but as `styleDefinition` is meant to hold all widgets styles it should contain also classes. While we can't get rid of `data.classes` it will be duplicated...